File: 43. Multiply Strings/C++/Multiply Strings.cpp
class Solution {
public:
    string multiply(string num1, string num2) {
        if (num1 == "0" || num2 == "0") {
            return "0";
        }
        vector<int> result(num1.length() + num2.length(), 0);
        for (int i = num1.length() - 1; i >= 0; i--) {
            for (int j = num2.length() - 1; j >= 0; j--) {
                int mul = (num1[i] - '0') * (num2[j] - '0');
                int pos1 = i + j;
                int pos2 = i + j + 1;
                mul += result[pos2];
                result[pos1] += mul / 10;
                result[pos2] = mul % 10;
            }
        }
        string str = "";
        if (result[0] != 0) {
            str += to_string(result[0]);
        }
        for (int i = 1; i < result.size(); i++) {
            str += to_string(result[i]);
        }
        return str;
    }
};

File: 043. Multiply Strings/Python/Multiply Strings.py
class Solution(object):
    def multiply(self, num1, num2):
        """
        :type num1: str
        :type num2: str
        :rtype: str
        """
        if num1 == '0' or num2 == '0':
            return '0'
        result = [0] * (len(num1) + len(num2))
        for i in range(len(num1) - 1, -1, -1):
            for j in range(len(num2) - 1, -1, -1):
                mul = int(num1[i]) * int(num2[j])
                pos_1 = i + j
                pos_2 = i + j + 1
                mul += result[pos_2]
                result[pos_1] += mul // 10
                result[pos_2] = mul % 10
        result = ''.join(str(item) for item in result)
        if result[0] == '0':
            result = result[1:]
        return result
